tests/network-ovn

```diff
@@ -2019,6 +2019,94 @@ ovn_l3only_tests() {
     lxc network delete lxdbr0
 }
 
+# countDynamicLeases is a helper function that counts dynamic leases of the instance within
+# all network leases.
+countDynamicLeases(){
+        network="$1"
+        instance="$2"
+        count=$(lxc network list-leases "${network}" --format compact | awk -v inst="${instance}" '$1 == inst && $4 == "DYNAMIC" { count++ } END { print count+0 }')
+        echo "${count}"
+}
+
+# countStaticLeases is a helper function that counts static leases of the instance within
+# all network leases.
+countStaticLeases(){
+        network="$1"
+        instance="$2"
+        count=$(lxc network list-leases "${network}" --format compact | awk -v inst="${instance}" '$1 == inst && $4 == "STATIC" { count++ } END { print count+0 }')
+        echo "${count}"
+}
+
+# hasLeaseIP is a helper function that checks whether the instance has the given static lease IP.
+hasLeaseIP(){
+        network="$1"
+        instance="$2"
+        ip="$3"
+        result=$(lxc network list-leases "${network}" --format compact | awk -v inst="${instance}" -v ip="${ip}" '$1 == inst && $3 == ip { print $3 }')
+        [ -z "${result}" ] && echo 1 || echo 0
+}
+
+ovn_leases_tests() {
+        lxc network create ovnbr0 \
+            ipv4.address=10.10.10.1/24 ipv4.nat=true \
+            ipv4.dhcp.ranges=10.10.10.2-10.10.10.199 \
+            ipv4.ovn.ranges=10.10.10.200-10.10.10.254 \
+            ipv6.address=fd42:4242:4242:1010::1/64 ipv6.nat=true \
+            ipv6.ovn.ranges=fd42:4242:4242:1010::200-fd42:4242:4242:1010::254
+
+        lxc network create test-ovn --type=ovn network=ovnbr0
+        lxc network set test-ovn \
+                ipv4.address=10.10.11.1/24 \
+                ipv6.address=fd42:1000:1000:1011::1/64
+
+        # Ensure the instance gets both ipv4 and ipv6 dynamic leases and that they
+        # do not disappear after restart.
+        lxc launch "${instanceImage}" c -s default -n test-ovn
+        [ "$(countDynamicLeases test-ovn c)" = "2" ]
+        [ "$(countStaticLeases test-ovn c)" = "0" ]
+
+        lxc restart c
+        [ "$(countDynamicLeases test-ovn c)" = "2" ]
+        [ "$(countStaticLeases test-ovn c)" = "0" ]
+
+        lxc rm --force c
```

Comment:
I find that sticking to primary action verbs (`delete` rather than the `rm` alias) makes it easier to track instance lifetime. Also, the test suite seems to consistently use `lxc delete -f c1`.